packages/server/graphql/mutations/loginWithGoogle.ts

```diff
@@ -86,22 +87,28 @@ const loginWithGoogle = {
         }
       }
 
+      const userId = `google-oauth2|${sub}`
+      const deletedUser = await r
+        .table('User')
+        .get(userId)
+        .run()
+
       // it's a new user!
       const nickname = name || email.substring(0, email.indexOf('@'))
       const preferredName = nickname.length === 1 ? nickname.repeat(2) : nickname
-      const userId = `google-oauth2|${sub}`
       const identity = new AuthIdentityGoogle({
         id: sub,
         isEmailVerified: email_verified !== 'false'
       })
       const newUser = new User({
-        id: userId,
```

Comment:
let's do the work up here so we can avoid mutating the object down below.
that way future us won't have to know that `newUser.id` is invalid when the object is instantiated
 
```
id: deletedUser?.isRemoved ? `google-oauth2|${generateUID()}` : userId
```